aptos-move/aptos-release-builder/src/simulate.rs

```diff
@@ -513,6 +554,9 @@ pub async fn simulate_multistep_proposal(
     for (script_idx, (script_path, (script_blob, script_hash))) in
         proposal_scripts.iter().zip(compiled_scripts).enumerate()
     {
+        // Force-end the epoch so that buffered configuration changes get applied.
+        force_end_epoch(&state_view).context("failed to force end epoch")?;
```

Comment:
In the case where we need to have those configuration loaded, shouldn't the script itself call a reconfiguration instead?

Reply:
This is no longer how things work with the introduction of DKG. Now when you call `reconfigure`, they just buffer all the upcoming changes. The changes will be applied when the next DKG event comes (next epoch).

Reply:
I thought reconfiguration still triggers new epoch, just ~30s delayed. I guess what we're doing here is to avoid waiting for that time. Would that causing any problem since this is not exactly mimic Mainnet anymore?

Reply:
So how did @lightmark's proposal execute on testnet then? Is it because we execute 1 script, and the second becomes retry due to reconfig, and we execute it in the new block when also new epoch possibly kicks in?

Reply:
> So how did @lightmark's proposal execute on testnet then? Is it because we execute 1 script, and the second becomes retry due to reconfig, and we execute it in the new block when also new epoch possibly kicks in?

Was asking the same question. I don't understand how our automated testnet deployment workflow handles this without issues. Does it have any sort of waiting period?